김가은/WEEK1.js

```diff
@@ -0,0 +1,31 @@
+// 최종 풀이 1
+function solution(s) {
+  let answer = 0;
+  let arr = ["()", "[]", "{}"];
+
+  for (let i = 0; i < s.length; i++) {
```

Comment:
```suggestion
  for (let i = 0; i < s.length; ++i) {
```

안녕하세요 지나가던 오지라퍼입니다

[후위 증감 보다는 전위 증감이 성능상 빠르다고 해요](https://www.inflearn.com/questions/171046) (물론 크리티컬하진 않습니다...) 

---

추가적으로 자바스크립트 반복문에 대한 퍼포먼스는 [다음 글](https://shlrur.github.io/develog/2018/10/18/about-loop-performance/)에서 확인하실 수 있슴다

> 물론 반복수가 엄청 높아지지 않으면 (최대 입력수) 유의미 하지는 않아, 가독성이 좋은 방법, 편하신 방법으로 하셔도 무관해요

---

질문에 남겨주셨던 스택 (배열)에 대한 부분 또한 제가 아는 선에서는 ... 시간 성능, 메모리 성능 둘 다 큰 의미는 없어 보입니다 (물론 제가 부족해서 못보는 것일 확률 99퍼센트입니다)

메모리 적으로 사용이 많이 된다고 생각하실 수도 있지만, v8 기반의 node 환경에서는 메모리 관리를 세대적 가설을 따르고 있기 때문에 큰 누수는 없을 것으로 생각됩니다

> 관련해서 정리한 글 홍보 https://www.hyesungoh.xyz/how-to-collect-garbage-in-js#%EC%84%B8%EB%8C%80%EC%A0%81-%EA%B0%80%EC%84%A4

(아니라면 말해주세요)

이상 오지라퍼였습니다 ... 죄송합니다 자바스크립트가 반가운 마음에 신나버렸습니다 ..

Reply:
헛 이렇게 답변을 해주시다니.. 감사합니다 😍 
자주 알감자 구경해주시고 코멘트 환영합니다..ㅎㅎ